apps/user/serializers.py

```diff
@@ -166,6 +166,28 @@ def validate(self, attrs):
         raise serializers.ValidationError(gettext('Activation link is not valid.'))
 
 
+class VerifySerializer(serializers.Serializer):
+    id = serializers.IntegerField(required=True, write_only=True)
+    is_verified = serializers.BooleanField()
+
+    def validate(self, attrs):
+        user_type = self.context['request'].user.user_type
+        if not user_type == User.UserType.MODERATOR:
+            raise serializers.ValidationError(gettext('Permission denied'))
+        id = attrs.get('id', None)
+        is_verified = attrs.get('is_verified', None)
+        try:
+            user = User.objects.get(pk=id)
+        except (TypeError, ValueError, OverflowError, User.DoesNotExist):
+            user = None
+
+        if user is not None:
+            user.is_verified = is_verified
+            user.save()
+            return attrs
+        raise serializers.ValidationError(gettext('User with this id doesnot exists'))
```

Comment:
Let's never change the database entity through validation.
We have basic mutation where you can specify custom permissions like this one. Add permission in configs/permissions.py VERIFY_USER, and then mention that permission using CreateUpdateGrapheneMutation with id. That way you don't have to retrieve the user yourself and validate user permissions.

https://github.com/kitab-bazar/server/blob/afbb95ec58e110851ca6196c8855d5aebcbb29bd/apps/order/mutations.py#L95-L102

```suggestion
    def validate(self, data):
        user_type = self.context['request'].user.user_type
        if not user_type == User.UserType.MODERATOR:
            raise serializers.ValidationError(gettext('Permission denied'))
        return data
```